src/Psy/Plugin/AbstractPlugin.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Psy\Plugin;
+
+abstract class AbstractPlugin
+{
+    public static function register()
+    {
+        Manager::register(new static(), static::getName());
+    }
+
+    /**
+     * @return string
+     */
+    final public static function getName()
+    {
+        $class = new \ReflectionClass(get_called_class());
+
+        return preg_replace('#Plugin$#', '', $class->getShortName());
+    }
+
+    /**
+     * @param array $configuration
+     *
+     * @return array
+     */
+    final public static function getConfiguration($configuration = array())
+    {
+        return array_merge_recursive(
+            $configuration,
+            array(
+               'commands'   => static::getCommands(),
+               'presenters' => static::getPresenters(),
+               'matchers'   => static::getMatchers(),
```

Comment:
It'd be super useful for plugins which depend on plugins. For example, Drupal, Laravel and EZ all depend on some parts of Symfony. So if there were a Symfony plugin, perhaps those others would want the Symfony commands, but not presenters or matchers. Or if there were a Doctrine plugin, I could see wanting the ability to present and match Doctrine models, but not expose all the Doctrine console commands, because they are legion.